core/src/HIP/Kokkos_HIP_ParallelReduce_MDRange.hpp

```diff
@@ -122,30 +122,36 @@ class ParallelReduce<CombinedFunctorReducerType,
         __syncthreads();
       }
 
-      for (unsigned i = threadIdx.y; i < word_count.value; i += blockDim.y) {
+      for (word_size_type i = threadIdx.y; i < word_count.value;
+           i += blockDim.y) {
         global[i] = shared[i];
       }
     }
   }
 
   // Determine block size constrained by shared memory:
-  // This is copy/paste from Kokkos_HIP_Parallel_Range
   inline unsigned local_block_size(const FunctorType& f) {
-    const auto& instance = m_policy.space().impl_internal_space_instance();
-    auto shmem_functor   = [&f](unsigned n) {
-      return hip_single_inter_block_reduce_scan_shmem<false, WorkTag,
-                                                      value_type>(f, n);
-    };
-
-    unsigned block_size =
-        Kokkos::Impl::hip_get_preferred_blocksize<ParallelReduce, LaunchBounds>(
-            instance, shmem_functor);
-    if (block_size == 0) {
+    unsigned n = HIPTraits::WarpSize * 8;
+    int const maxShmemPerBlock =
+        m_policy.space().hip_device_prop().sharedMemPerBlock * 0.95;
```

Comment:
Can you provide a comment on where 0.95 is coming from?

Reply:
but the static memory should be available from the functor properties or?